discount_code.go

```diff
@@ -0,0 +1,84 @@
+package goshopify
+
+import (
+	"fmt"
+	"time"
+)
+
+const discountCodeBasePath = "/admin/price_rules/%d/discount_codes"
+
+// DiscountCodeService is an interface for interfacing with the discount endpoints
+// of the Shopify API.
+// See: https://help.shopify.com/en/api/reference/discounts/PriceRuleDiscountCode
+type DiscountCodeService interface {
+	Create(int, PriceRuleDiscountCode) (*PriceRuleDiscountCode, error)
+	Update(int, PriceRuleDiscountCode) (*PriceRuleDiscountCode, error)
+	List(int) ([]PriceRuleDiscountCode, error)
+	Get(int, int) (*PriceRuleDiscountCode, error)
+	Delete(int, int) error
+}
+
+// DiscountCodeServiceOp handles communication with the blog related methods of
```

Comment:
I think this "blog related methods" bit should be changed to reflect the actual use of this struct